docs/source/topics/proc_accessing-the-internal-openshift-registry.adoc

```diff
@@ -0,0 +1,92 @@
+[id="accessing-the-internal-openshift-registry_{context}"]
+= Accessing the internal OpenShift registry
+
+The OpenShift cluster running in the {prod} virtual machine includes an internal container image registry by default.
+This internal container image registry can be used as a publication target for locally developed container images.
+To access the internal OpenShift registry, follow these steps.
+
+.Prerequisites
+
+* A running {prod} virtual machine.
+For more information, see link:{crc-gsg-url}#starting-the-virtual-machine_gsg[Starting the virtual machine].
+* A working [command]`oc` command.
+For more information, see link:{crc-gsg-url}#accessing-the-openshift-cluster-with-oc_gsg[Accessing the OpenShift cluster with `oc`].
+* For Docker, add `default-route-openshift-image-registry.apps-crc.testing` as an insecure registry.
+For more information, see link:https://docs.docker.com/registry/insecure/[the Docker documentation].
+
+.Procedure
+
+. Get the route to the internal OpenShift registry:
++
+[subs="+quotes,attributes"]
+----
+$ oc get route -n openshift-image-registry
+----
+
+. Check which user is logged in to the cluster:
++
+[subs="+quotes,attributes"]
+----
+$ oc whoami
+----
++
+[NOTE]
+====
+For demonstration purposes, the current user is assumed to be `kube:admin`.
+====
+
+. Log in to the registry as that user with its token:
++
+[subs="+quotes,attributes"]
+----
+$ podman login -u kubeadmin -p $(oc whoami -t) default-route-openshift-image-registry.apps-crc.testing --tls-verify=false
+----
+
+. Create a new project:
++
+[subs="+quotes,attributes"]
+----
+$ oc new-project demo
+----
+
+. Pull an example container image:
++
+[subs="+quotes,attributes"]
+----
+$ podman pull quay.io/libpod/alpine
+----
+
+. Tag the image, including namespace details:
++
+[subs="+quotes,attributes"]
+----
+$ podman tag alpine:latest default-route-openshift-image-registry.apps-crc.testing/demo/alpine:latest
+----
+
+. Push the container image to the internal registry:
++
+[subs="+quotes,attributes"]
+----
+$ podman push default-route-openshift-image-registry.apps-crc.testing/demo/alpine:latest
+----
+
+. Get imagestreams and verify that the pushed image is listed:
++
+[subs="+quotes,attributes"]
+----
+$ oc get is
+----
+
+. Enable image lookup for all tags in the imagestream:
```

Comment:
no idea what this means? Is there some relevant openshift documentation this could link to?

Reply:
I'm not entirely certain either, haha. If I understand it correctly, this means that we can essentially just use the image name/tag instead of having to fully qualify it with the URL for the internal registry, but I may be incorrect about that.

I searched the docs for OpenShift 4.x and could only find this referenced in relation to using Kubernetes resources in OpenShift. That didn't really assist with my understanding of this option.

@praveenkumar, would you mind explaining this a bit more for us? Also, is this a necessary step for the procedure, or is it a "nice to have" for making the images easier to work with?

Reply:
https://docs.openshift.com/container-platform/4.6/openshift_images/using-imagestreams-with-kube-resources.html one might help

Reply:
Found in the man page for `oc-set-image-lookup`: "Local name lookup allows an image stream to be the source of images for pods, deployments, replica sets, and other resources that reference images, without having to provide the full registry URL."